### PR TITLE
cleanup rename p to wle

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go [hunk 1]
```diff
@@ -87,8 +87,8 @@ func (a *index) WorkloadsCollection(
 		}
 		services := []model.ServiceInfo{*svc}
 
-		for _, p := range se.Spec.Endpoints {
-			meshCfg := krt.FetchOne(ctx, MeshConfig.AsCollection())
+		meshCfg := krt.FetchOne(ctx, MeshConfig.AsCollection())
```
Comment (MorrisLaw) [inline pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go:90]: Nice

File: pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go [hunk 2]
```diff
@@ -99,10 +99,10 @@ func (a *index) WorkloadsCollection(
 				return t.ResourceName()
 			}))
 			// We could do a non-FilterGeneric but krt currently blows up if we depend on the same collection twice
-			auths := fetchPeerAuthentications(ctx, PeerAuths, meshCfg, se.Namespace, p.Labels)
+			auths := fetchPeerAuthentications(ctx, PeerAuths, meshCfg, se.Namespace, wle.Labels)
 			policies = append(policies, convertedSelectorPeerAuthentications(meshCfg.GetRootNamespace(), auths)...)
 			var waypoint *Waypoint
-			if p.Labels[constants.ManagedGatewayLabel] != constants.ManagedGatewayMeshControllerLabel {
+			if wle.Labels[constants.ManagedGatewayLabel] != constants.ManagedGatewayMeshControllerLabel {
 				// Waypoints do not have waypoints, but anything else does
```
Comment (MorrisLaw) [inline pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go:106]: What are we referring to when we say anything?

**Review by MorrisLaw — COMMENTED** [review]
Good refactor! Are there any tests that are related to this code, that we might need to also refacfor for consistency?

File: pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go (hunk 2 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go:106]: This comments make no sense at this place. Actually it means service/pod/wle/namespace can have waypoint capturred

**Review by howardjohn — APPROVED** [review]

**Pull request merged**